pkgs/by-name/ri/rifiuti/package.nix

```
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, fetchzip
+, fetchpatch
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "rifiuti";
+  version = "20040505_1";
+
+  src = fetchzip {
+    url = "mirror://sourceforge/project/odessa/Rifiuti/${finalAttrs.version}/rifiuti_${finalAttrs.version}.zip";
+    hash = "sha256-bVPz0nXexGtQtXxGb3Mc79tzyZikc7KxNNWuvgu6pQ0=";
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://salsa.debian.org/pkg-security-team/rifiuti/-/raw/f237358a91b12776beb9942c79ccb3aea180968a/debian/patches/add-GCC-hardening";
+      hash = "sha256-4cQSNtnHRD88TEP2iESgFlY16Y8WZZityIQh+M1kchQ=";
+    })
```

Consider adding the name attribute like so:
```suggestion
    (fetchpatch {
      name = "add-GCC-hardening.patch";
      url = "https://salsa.debian.org/pkg-security-team/rifiuti/-/raw/f237358a91b12776beb9942c79ccb3aea180968a/debian/patches/add-GCC-hardening";
      hash = "sha256-4cQSNtnHRD88TEP2iESgFlY16Y8WZZityIQh+M1kchQ=";
    })
```

---

`fetchpatch` is only needed over `fetchurl` when the patch is dynamically generated from a source control system. This seems to be a static file so `fetchurl` should be fine.

---

Fixed